app/controllers/listings_controller.rb

```diff
@@ -21,6 +21,16 @@ def create
     end
   end
 
+  def edit
+    @listing = Listing.find(params[:id])
+  end
+
+  def update
+    @listing = Listing.find(params[:id])
+    @listing = Listing.update(listing_params)
```

Comment:
Line 30 should be `@listing.update(listing_params)`

Reply:
It works also this way, but I can change it